### PR TITLE
Revive classes

### LINKED ISSUE #34: Add support for serialising ES6 classes
We could try serializing classes similar to [typestack/class-transformer](https://github.com/typestack/class-transformer).

### PR DESCRIPTION
**Pull request by Skn0tt**
Closes #34 

Implements revival of ES6 classes by attaching the original prototypes during revival. It requires registering all neccessary classes using `SuperJSON.registerClass(...)`.
In the future, this could be done automatically in a compile-step (Which poses it's own issues, e.g. figuring out which classes are actually used and which classes to bundle therefore, but would make it opaque to Blitz users).

### CODE REVIEW

**Comment by Skn0tt** [comment]
One thing I couldn't figure out is how to extend `SuperJSONValue` to any class-based values - do you have an idea on this @merelinguist? See the type errors introduced in https://github.com/blitz-js/superjson/pull/36/commits/70f2912aa725033025810295b2f5ed1a694449bd for reference.

File: src/transformer.ts [hunk 1]
```diff
@@ -103,37 +103,80 @@ export const transformValue = (
...
+      }
+
+      case 'class': {
+        const clazz = ClassRegistry.getClass(type[1]);
+
+        if (!clazz) {
+          throw new Error('Trying to deserialize unknown class');
+        }
+
+        return Object.assign(Object.create(clazz.prototype), json);
```
Comment (Skn0tt) [inline src/transformer.ts:175]: This is where the magic happens btw. `Object.create(clazz.prototype)` creates a new, empty instance of `clazz`, which [is much faster](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/setPrototypeOf) than changing the prototype of the existing value using `Object.setPrototypeOf`

**Comment by merelinguist** [comment]
@Skn0tt Hmm don’t really have any ideas about the types. Is it mergeable with looser typing?

**Comment by Skn0tt** [comment]
@merelinguist this is the best I could come up with: 29d10cc. This will effectively downgrade our types to `any`, but I don't see any way to make it better :(

**Pull request merged**